x-pack/plugins/encrypted_saved_objects/server/saved_objects/encrypted_saved_objects_client_wrapper.ts

```diff
@@ -217,15 +232,27 @@ export class EncryptedSavedObjectsClientWrapper implements SavedObjectsClientCon
    * Strips encrypted attributes from any non-bulk Saved Objects API response. If type isn't
    * registered, response is returned as is.
    * @param response Raw response returned by the underlying base client.
+   * @param [originalAttributes] Optional list of original attributes of the saved object.
+   * @param [namespace] Optional namespace that was used for the saved objects operation.
    */
-  private stripEncryptedAttributesFromResponse<T extends SavedObjectsUpdateResponse | SavedObject>(
-    response: T
-  ): T {
-    if (this.options.service.isRegistered(response.type) && response.attributes) {
-      response.attributes = this.options.service.stripEncryptedAttributes(
-        response.type,
-        response.attributes as Record<string, unknown>
+  private async handleEncryptedAttributesInResponse<
+    T,
+    R extends SavedObjectsUpdateResponse<T> | SavedObject<T>
+  >(response: R, originalAttributes?: T, namespace?: string): Promise<R> {
+    if (response.attributes && this.options.service.isRegistered(response.type)) {
+      const { attributes, error } = await this.options.service.stripOrDecryptAttributes(
+        { id: response.id, type: response.type, namespace },
+        response.attributes as Record<string, unknown>,
+        originalAttributes as Record<string, unknown>,
+        // If we cannot decrypt encrypted attributes let's strip them off and record the error to
+        // let consumer decide whether to fail or handle recovery gracefully.
+        { stripOnDecryptionError: true }
       );
+
+      response.attributes = attributes as T;
+      if (error) {
+        response.error = SavedObjectsErrorHelpers.decorateGeneralError(error) as any;
```

Comment:
__note:__ currently we don't have cases when both `attributes` and `error` are returned at the same time, but the use case we have here feels like a good precedent to return both, what do you think?

Reply:
Agree